runtime/gc_base/GCExtensions.cpp

```diff
@@ -235,6 +236,24 @@ MM_GCExtensions::updateIdentityHashDataForSaltIndex(UDATA index)
 	getJavaVM()->identityHashData->hashSaltTable[index] = (U_32)convertValueToHash(getJavaVM(), getJavaVM()->identityHashData->hashSaltTable[index]);
 }
 
+BOOLEAN
+MM_GCExtensions::isOriginalJDK8HeapSizeCompatibilityOptionsEanbled(MM_EnvironmentBase* env)
+{
+	J9JavaVM* vm = (J9JavaVM *)getOmrVM()->_language_vm;
+	BOOLEAN enabled = FALSE;
+	/* Java 11 or above would ignore VMOPT_XXENABLEORIGINALJDK8HEAPSIZECOMPATIBILITY option */
+	if (J2SE_VERSION((J9JavaVM *)getOmrVM()->_language_vm) < J2SE_V11) {
+		IDATA argIndex = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXENABLEORIGINALJDK8HEAPSIZECOMPATIBILITY, NULL);
+		if (-1 != argIndex) {
+			IDATA argIndex2 = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXDISABLEORIGINALJDK8HEAPSIZECOMPATIBILITY, NULL);
+			if (argIndex > argIndex2) {
+				enabled = TRUE;
+			}
+		}
+	}
+	return enabled;
```

Comment:
This code can be organized better. There is an example from jvminit.c (you don't need `else if` case):
```
	{
		IDATA enablePositiveHashCode = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXENABLEPOSITIVEHASHCODE, NULL);
		IDATA disablePositiveHashCode = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXDISABLEPOSITIVEHASHCODE, NULL);
		if (enablePositiveHashCode > disablePositiveHashCode) {
			vm->extendedRuntimeFlags |= J9_EXTENDED_RUNTIME_POSITIVE_HASHCODE;
		} else if (enablePositiveHashCode < disablePositiveHashCode) {
			vm->extendedRuntimeFlags &= ~(UDATA)J9_EXTENDED_RUNTIME_POSITIVE_HASHCODE;
		}
	}
```